lib/checking_account.rb

```diff
@@ -0,0 +1,50 @@
+module Bank
+
+  class CheckingAccount < Account
+
+    def initialize(id, initial_balance, open_date)
+      super
+      @fee = 1.0
+      @check_number = 1
+    end
+
+    # CheckingAccount uses the withdraw method from Account
+    # This method does not need to be called here, but is left in for clarity
+    def withdraw(amount_to_withdraw)
+      super
+    end
+
+    def withdraw_using_check(amount_to_withdraw)
+      initial_balance = @balance
+      # min_balance is set to -10 for check withdrawals
+      @min_balance = -10
+      # If the customer has used fewer than 3 checks, the fee is set to 0, and the withdraw method from Account is run
+      if @check_number <= 3
+        @fee = 0
+        withdraw(amount_to_withdraw)
```

Comment:
I like that you're utilizing the existing withdraw method to take care of this functionality